php/class-utils.php

```diff
@@ -710,8 +713,8 @@ public static function purge_fragments() {
 	 */
 	public static function purge_fragment( $index ) {
 		if ( isset( self::$file_fragments[ $index ] ) ) {
-			fclose( self::$file_fragments[ $index ] ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fclose
-			unset( self::$file_fragments[ $index ] );
+			fclose( self::$file_fragments[ $index ]['pointer'] ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fclose
```

Comment:
Why are we avoiding the WP_Filesystem?

Reply:
That's a great question @yuval-cloudinary.

The single reason is VIP's filesystem. Please [learn more](https://docs.wpvip.com/vip-file-system/media-uploads/#h-wp-filesystem-api) here.

Reply:
I guess you mean the note "most WP_Filesystem operations related to directories will not work as expected.", is it really an issue here?
If yes, does it hurt us by not being able to serve some customers, and compromise security for others?
Maybe we can detect if VIP is used, and decide if accordingly?